### PR TITLE
Debug gate cloning

### PR DESCRIPTION
**Pull request by openroadie**
My local results don't show any improvement with gate cloning on but don't show degradation either. The artificial testcase does show improvement.

### CODE REVIEW

**Comment by openroadie** [comment]
ORFS testing results should be available here:

https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1196

File: src/rsz/test/repair_fanout7.ok [hunk 1]
```diff
@@ -11,7 +11,8 @@ load DFF_X1 CK D 300000 metal3 1000
...
 [INFO ODB-0133]     Created 4 nets and 307 connections.
 worst slack -23.63
-[INFO RSZ-0043] Swapped pins on 1 instances.
-[INFO RSZ-0049] Cloned 1 instances.
+[INFO RSZ-0040] Inserted 34 buffers.
+[INFO RSZ-0041] Resized 139 instances.
+[INFO RSZ-0049] Cloned 6 instances.
 [WARNING RSZ-0062] Unable to repair all setup violations.
-worst slack -0.06
+worst slack -3.23
```
Comment (maliberty) [inline src/rsz/test/repair_fanout7.ok:18]: Is this degradation expected?

**Review by maliberty — COMMENTED** [review]
The description says this is a draft pr for testing. Are you planning to merge this?

File: src/rsz/test/repair_fanout7.ok (hunk 1 above)
Comment (openroadie) [reply src/rsz/test/repair_fanout7.ok:18]: The earlier result was not correct. The comparison is with repair_fanout8 (which shows something like -5 in wns). So yes. This degradation can be waived.

**Comment by openroadie** [comment]
> The description says this is a draft pr for testing. Are you planning to merge this?

Fixed. This is ready for merge. ORFS shows no issues (see link above).

File: src/rsz/src/RepairSetup.cc [hunk 2]
```diff
@@ -656,6 +641,127 @@ RepairSetup::upsizeCell(LibertyPort *in_port,
...
+
+  LibertyCell *original_cell = network_->libertyCell(drvr_inst);
+  LibertyCell *clone_cell = resizer_->halfDrivingPowerCell(original_cell);
+
+  if (clone_cell == nullptr) {
+    clone_cell = original_cell; // no clone available use original
+  }
+  // TODO This location can be optimized as the centroid of all the pins we
+  // are connecting to the output of the buffer and the input pins of the gate
+  Point drvr_loc = db_network_->location(drvr_pin);
```
Comment (maliberty) [inline src/rsz/src/RepairSetup.cc:706]: Without this TODO I'm not surprised there isn't much improvement.

**Review by maliberty — APPROVED** [review]

**Pull request merged**